src/Symfony/Component/HttpClient/DataCollector/HttpClientDataCollector.php

```diff
@@ -250,13 +237,8 @@ private function getCurlCommand(array $trace): ?string
         return implode(" \\\n  ", $command);
     }
 
-    /**
-     * Let's be defensive : we authorize only size of 8kio on Windows for escapeshellarg() argument to avoid a fatal error.
-     *
-     * @see https://github.com/php/php-src/blob/9458f5f2c8a8e3d6c65cc181747a5a75654b7c6e/ext/standard/exec.c#L397
-     */
-    private function argMaxLengthIsSafe(string $payload): bool
+    private function sanitizePayload(string $payload): string
     {
-        return \strlen($payload) < ('\\' === \DIRECTORY_SEPARATOR ? 8100 : 256000);
+        return "'".str_replace("'", "'\\''", $payload)."'";
```

Comment:
This is broken on Windows, where single quotes are not valid shell quoting.

Reply:
Nicolas suggested to do this, even for Windows: https://github.com/symfony/symfony/issues/49693#issuecomment-1754694141. However, I can update this bit if you think this is necessary

Reply:
I updated the method to work on Windows using double quotes when necessary